### ISSUE TITLE
Invalid environment variable names for `Command` are not rejected

### ISSUE THREAD

**Issue by dylni** [rust-lang/rust#101978]
I tried this code:

```rust
use std::process::Command;

fn main() {
    println!(
        "{:?}",
        Command::new("printenv")
            .arg("foo")
            .env("foo=bar", "baz")
            .output()
    );
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f3e035599c38c2bea88c8331a0b94da1

I expected to see this happen: an error that the environment variable name contains `=`

Instead, this happened:

```
Ok(Output { status: ExitStatus(unix_wait_status(0)), stdout: "bar=baz\n", stderr: "" })
```

There was a PR for to give errors for these variables, but it was closed due to inactivity: #39338

### Meta

`rustc --version`:
```
rustc 1.63.0 (4b91a6ea7 2022-08-08)
```

**Comment by chenyukang** [comment]
Is there any standards explain why we need to reject `=` in environment variable names?

I tried in local, seems Bash accept it:

```console
cat@LAPTOP-V6U0QKD4:~/code/rust$ foo=bar=baz printenv foo
bar=baz
```

**Comment by dylni** [comment]
@chenyukang The current behavior is unexpected because `.env("foo=bar", "baz")` will not set an environment variable called `foo=bar`. It will set a variable called `foo`.

For example, this call will fail because `foo=bar` is not a valid variable name:

```
env::set_var("foo=bar", "baz");
```

**Comment by RodBurman** [comment]
I tried the following in /bin/zsh and also sh

% export "foo=bar"=baz
and
% printenv foo=bar    
baz

So you can have an environmental variable called foo=bar (at least in macOS) and so Command is correct I believe.
However the following code does not work:
```
fn main() {
    std::env::set_var("\"foo=bar\"", "baz");
 }
```
It compiles but running it fails with:

```
thread 'main' panicked at /Users/rod/.rustup/toolchains/stable-aarch64-apple
darwin/lib/rustlib/src/rust/library/std/src/env.rs:364:9:
failed to set environment variable `"\"foo=bar\""` to `"baz"`: Invalid argument (os error 22)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

If the extra quotes are removed (so "foo1=bar1") it fails similarly as do single quote or using an os_str. I am not sure the environmental variable foo=bar can be set from a rust programme unlike using _export_.
